gui/elements/qt/_control_bars.py

```diff
@@ -0,0 +1,59 @@
+from PyQt5.QtWidgets import QHBoxLayout, QWidget
+from PyQt5.QtCore import QSize
+
+from ._range_slider import QVRangeSlider
+
+
+class QtControlBars(QWidget):
+    def __init__(self, control_bars):
+        super().__init__()
+
+        self.control_bars = control_bars
+
+        layout = QHBoxLayout()
+
+        # Clim Slider
+        self.climSlider = QVRangeSlider(slider_range=[0, 1, 0.0001],
+                                        values=[0, 1], parent=self)
+        self.climSlider.setEmitWhileMoving(True)
+        self.climSlider.collapsable = False
+        layout.addWidget(self.climSlider)
+
+        self.setLayout(layout)
+        self.setMouseTracking(True)
+        self.setMinimumSize(QSize(40, 40))
+
+        self.climSlider.rangeChanged.connect(self.clim_slider_changed)
+        self.control_bars.events.update_slider.connect(self.update)
+
+    def update(self, event):
+        if event.enabled:
+            self.climSlider.setEnabled(True)
+            self.climSlider.setValues(event.values)
+        else:
+            self.climSlider.setEnabled(False)
+
+    def clim_slider_changed(self, slidermin, slidermax):
+        self.control_bars.values = (slidermin, slidermax)
+        msg = None
+        for layer in self.control_bars.viewer.layers:
+            if hasattr(layer, 'visual') and layer.selected:
+                valmin, valmax = layer._clim_range
+                cmin = valmin+self.control_bars.values[0]*(valmax-valmin)
+                cmax = valmin+self.control_bars.values[1]*(valmax-valmin)
+                layer.clim = [cmin, cmax]
+                msg = '(%.3f, %.3f)' % (cmin, cmax)
+        if msg is None:
+            msg = 'Ready'
+
+        self.control_bars.viewer.status = msg
+
+    def mouseMoveEvent(self, event):
+        for layer in self.control_bars.viewer.layers[::-1]:
```

Comment:
Wait, why is this iteration backwards?

Reply:
Iteration is going backwards as we are looking for the "highest" image layer (i.e. the one on top, i.e. the one visible) - this corresponds to the most recently added image layer, which will be the first one we encounter when iterating through the list backwards. The alternative would be to build the list by adding each subsequent image layer at the beginning not the end. This would cause the index of all the image layers to change anytime a new one is added. I think it is ok how it is now, but curious on your take?

Reply:
Hmm, a good point, appending is most natural. Maybe add a comment to this effect above this line?

Reply:
will do